terraform/ecs.tf

```diff
@@ -0,0 +1,182 @@
+  }
+}
+resource "aws_ecs_task_definition" "main" {
+  network_mode             = "awsvpc"
+  family                   = "terraformlearning"
+  requires_compatibilities = ["FARGATE"]
+  cpu                      = 256
+  memory                   = 512
+  execution_role_arn       = aws_iam_role.ecs_task_execution_role.arn
+  task_role_arn            = aws_iam_role.ecs_task_role.arn
+  container_definitions = jsonencode([{
+    name      = "${var.name}-container-${var.environment}"
+    image     = "${var.container_image}:latest"
+    essential = true
+    //environment = var.environment
+    portMappings = [{
+      protocol      = "tcp"
+      containerPort = var.container_port
+      hostPort      = var.container_port
+    }]
+    logConfiguration = {
+      logDriver = "awslogs"
+      options = {
+        awslogs-group         = "${var.name}-ecs-logs-${var.environment}"
+        awslogs-region        = "us-east-2",
+        awslogs-stream-prefix = "ecs"
+      }
+    }
+  }])
+}
+
+resource "aws_iam_role" "ecs_task_role" {
+  name = "${var.name}-ecsTaskRole"
+
+  assume_role_policy = <<EOF
+{
+ "Version": "2012-10-17",
+ "Statement": [
+   {
+     "Action": "sts:AssumeRole",
+     "Principal": {
+       "Service": "ecs-tasks.amazonaws.com"
+     },
+     "Effect": "Allow",
+     "Sid": ""
+   }
+ ]
+}
+EOF
+}
+
+resource "aws_iam_policy" "dynamodb" {
+  name        = "${var.name}-task-policy-dynamodb"
+  description = "Policy that allows access to DynamoDB"
+
+  policy = <<EOF
+{
+   "Version": "2012-10-17",
+   "Statement": [
+       {
+           "Effect": "Allow",
+           "Action": [
+               "dynamodb:CreateTable",
+               "dynamodb:UpdateTimeToLive",
+               "dynamodb:PutItem",
+               "dynamodb:DescribeTable",
+               "dynamodb:ListTables",
+               "dynamodb:DeleteItem",
+               "dynamodb:GetItem",
+               "dynamodb:Scan",
+               "dynamodb:Query",
+               "dynamodb:UpdateItem",
+               "dynamodb:UpdateTable"
+           ],
+           "Resource": "*"
+       }
+   ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy_attachment" "ecs-task-role-policy-attachment" {
+  role       = aws_iam_role.ecs_task_role.name
+  policy_arn = aws_iam_policy.dynamodb.arn
+}
+resource "aws_iam_role" "ecs_task_execution_role" {
+  name = "${var.name}-ecsTaskExecutionRole"
+
+  assume_role_policy = <<EOF
+{
+ "Version": "2012-10-17",
+ "Statement": [
+   {
+     "Action": "sts:AssumeRole",
+     "Principal": {
+       "Service": "ecs-tasks.amazonaws.com"
+     },
+     "Effect": "Allow",
+     "Sid": ""
+   }
+ ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy_attachment" "ecs-task-execution-role-policy-attachment" {
+  role       = aws_iam_role.ecs_task_execution_role.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy"
+}
+
+data "aws_subnet_ids" "public_subnet_ids" {
+  vpc_id = aws_vpc.main.id
```

Comment:
I'm not 100% sure what you're doing with these data blocks, but you shouldn't need them I don't think. [id](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/subnet#id) is exported attribute of the subnet resource, so you should be able to just use the resource directly.

Reply:
If I didn't use the data blocks I couldn't get access to the ID, something about it's not created yet.  I could just be doing something very, very wrong!

Reply:
Yeah that makes sense, I realized what it was for after. This way works, the disadvantage however is you'll likely hit a race condition when you spin up a new environment with this. Basically, this data block could search for your subnets before they're created down below.

Comment [here](https://github.com/theothermattm/kotlin-jvm-jargon-generator/pull/2#discussion_r633700828) should fix that. I added it a little late 🙂.